docs/source/development_release.rst

```diff
@@ -0,0 +1,80 @@
+.. _nbconvert_release:
+
+Making an ``nbconvert`` release
+===============================
+
+This document guides a contributor through creating a release of ``nbconvert``.
+
+
+Assign all merged PRs to milestones
+-----------------------------------
+
+Go to GitHub and assign all PRs that have been merged to milestones. 
```

Comment:
We can provide a handy link here to PRs merged with no milestone:
https://github.com/jupyter/nbconvert/pulls?utf8=%E2%9C%93&q=is%3Amerged%20is%3Apr%20no%3Amilestone%20